test/vsphere/e2e.sh

```diff
@@ -0,0 +1,172 @@
+  sudo chmod +x /usr/local/bin/sops
+}
+
+decrypt_vpn_certs() {
+  export AWS_ACCESS_KEY_ID=$SOPS_AWS_ACCESS_KEY_ID
+  export AWS_SECRET_ACCESS_KEY=$SOPS_AWS_SECRET_ACCESS_KEY
+
+  for filename in "$CERTS"/*.enc; do
+    # save the decrypted certs in the same folder w/o the ext .enc
+    sops --config "$PWD"/test/vsphere/.sops.yaml -d \
+      --input-type binary --output-type binary "$filename" >"${filename%.*}"
+  done
+
+  # clean a WARNING in openvpn
+  # WARNING: file '/openvpn/certs/user.key' is group or others accessible
+  chmod 400 "$CERTS"/user.key
+}
+
+dump_logs() {
+  mkdir -p "$ARTIFACTS"/snapshot
+  platform-cli snapshot --output-dir "$ARTIFACTS"/snapshot
+}
+
+on_exit() {
+  # dump the logs into the ARTIFACTS directory
+  dump_logs
+
+  # remove the cluster
+  # shellcheck disable=SC2086
+  platform-cli cleanup $PLATFORM_OPTIONS_FLAGS
+
+  # kill the VPN
+  docker kill vpn
+
+  # clean certs dir - remove all except enc
+  rm "$CERTS"/{*.crt,*.key}
+}
+
+wait_for_vpn() {
+  while [ -n "$(ip addr show tun0 2>&1 >/dev/null)" ]; do
+    sleep 0.1
+  done
+}
+
+generate_cluster_id() {
+  local prefix
+
+  prefix=$(tr </dev/urandom -cd 'a-f0-9' | head -c 5)
+  echo "e2e-${prefix}"
+}
+
+# These are used inside e2e.yml
+PLATFORM_CLUSTER_ID=$(generate_cluster_id)
+export PLATFORM_CLUSTER_ID
+export PLATFORM_CA=$PWD/.certs/ca.crt.pem
+export PLATFORM_PRIVATE_KEY=$PWD/.certs/ca.key.pem
+export PLATFORM_CA_CEK=foobar
+export PLATFORM_OPTIONS_FLAGS="-e name=${PLATFORM_CLUSTER_ID} -e domain=${PLATFORM_CLUSTER_ID}.lab.flanksource.com -vv"
+
+export PLATFORM_CONFIG=$PWD/test/vsphere/e2e.yml
+
+# These variables are required to configure access to vSphere, they also work with govc
+export GOVC_FQDN="vcenter.lab.flanksource.com"
+export GOVC_DATACENTER="lab"
+export GOVC_CLUSTER="cluster"
+export GOVC_NETWORK="VM Network"
+export GOVC_PASS="${GOVC_PASS}"
+export GOVC_USER="${GOVC_USER}"
+export GOVC_DATASTORE="datastore1"
+export GOVC_INSECURE=1
+export GOVC_URL="$GOVC_USER:$GOVC_PASS@$GOVC_FQDN"
+
+trap on_exit EXIT
+
+install_platformcli
+
+install_sops
+
+install_kubectl
+
+# Decrypt certs encrypted with SOPS and KMS Key from https://github.com/flanksource/vsphere-lab/blob/master/.sops.yaml
+decrypt_vpn_certs
+
+# Run the vpn client in container
+docker run --rm -d --name vpn -v "${PWD}/test/vsphere/openvpn/:/openvpn/" \
+  -w "/openvpn/" --cap-add=NET_ADMIN --net=host --device=/dev/net/tun \
+  gcr.io/cluster-api-provider-vsphere/extra/openvpn:latest
+
+echo "nameserver 10.255.0.2" | sudo tee -a /run/resolvconf/resolv.conf >/dev/null
+
+## Wait for VPN
+wait_for_vpn
+
+# Tail the vpn logs
+docker logs vpn
+
+# Generate Ingress certs
+platform-cli ca generate --name ingress-ca \
+  --cert-path "$PLATFORM_CA" --private-key-path "$PLATFORM_PRIVATE_KEY" \
+  --password "$PLATFORM_CA_CEK" --expiry 1
+mkdir -p .bin
+if [[ "$CIRCLECI" == "true" ]]; then
+  # .bin directory is owned by root for some reason
+  sudo chown circleci:circleci .bin
```

Comment:
Oh right know the reason of that and I know how to fix it.`.bin` folder is generated from the container during the installation of the binary. It is enough then to pass the circleci user to the container so make sure that the folder will have the right permissions 

Something like 
```
docker run --user $(uid)
```

Reply:
Actually seems not so trivial thing
```
failed to initialize build cache at /.cache/go-build: mkdir /.cache: permission denied
make: *** [Makefile:11: setup] Error 1
./test/vsphere/e2e.sh: line 69: platform-cli: command not found

Exited with code exit status 127
```

Reply:
https://github.com/docker-library/golang/issues/225#issuecomment-403170792

Actually the fix is quite easy. need to give a try of that.

Reply:
Fixed here all the permissions issue caused by docker volumes mount.

Reply:
Thank you @moshloop for taking a look at this